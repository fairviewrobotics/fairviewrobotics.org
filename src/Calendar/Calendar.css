/* TODO: create basic css for pages */
.page-title {
  position: absolute;
  font-size: 10vw;
  font-weight: 100;
  text-align: center;
  display: block;
  left: 0;
  right: 0;
  top: calc(5vw + 50px);
  color: var(--main-color);
}

.main-h2 {
  margin-top: 100px;
  margin-bottom: 1vw;
  font-size: 2.75em;
  width: 100%;
  padding: 0.25em;
  background-color: #eee;
  box-shadow: 0 2px 2px rgba(0, 0, 0, 0.2);
  color: #000;
  font-weight: 100;
}

.calendar {
  border-width: 0;
  box-shadow: 0 0 5px rgba(0, 0, 0, .3);
}

.calendar-container {
  margin-top: -100px;
}

.calendar-body {
  margin-top: 30vw;
  text-align: center;
}

@media(max-width:768px) {
  .calendar-body {
    margin-top: 40vw;
  }
}