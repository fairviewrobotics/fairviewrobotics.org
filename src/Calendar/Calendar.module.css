.header {
  margin-top: 100px;
  margin-bottom: 1vw;
  font-size: 2.75em;
  width: 100%;
  padding: 0.25em;
  background-color: #eee;
  box-shadow: 0 2px 2px rgba(0, 0, 0, 0.2);
  color: #000;
  font-weight: 100;
}

.calendar {
  border-width: 0;
  box-shadow: 0 0 5px rgba(0, 0, 0, .3);
}

.calendarContainer {
  margin-top: -100px;
  padding-top: 1.5em;
}

.calendarBody {
  text-align: center;
}