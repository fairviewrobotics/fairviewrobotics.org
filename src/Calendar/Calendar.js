import React, { PureComponent } from 'react';
import BackgroundImage from "../BackgroundImage/BackgroundImage";
import { Clearfix, Grid } from "react-bootstrap";

import './Calendar.css';

import intro6 from '../images/intro/6.jpg';

export default class Calendar extends PureComponent {
  render() {
    return (
      <div>
        <BackgroundImage src={intro6}/>
        <h1 className="page-title">Calendar</h1>
        <div className="calendar-body">
          <div className="calendar-section">
            <Grid bsClass="container" className="calendar-container">
              <h2 className="main-h2">Upcoming Events</h2>
              <Clearfix visibleXsBlock>
                <iframe
                  title="extra small calendar"
                  src="https://www.google.com/calendar/embed?showTitle=0&amp;showNav=0&amp;showDate=0&amp;showPrint=0&amp;showTabs=0&amp;showCalendars=0&amp;showTz=0&amp;mode=AGENDA&amp;height=600&amp;wkst=1&amp;bgcolor=%23FFFFFF&amp;src=fairviewrobotics%40gmail.com&amp;color=%23060D5E&amp;ctz=America%2FDenver"
                  className="calendar" width="100%" height="300" frameBorder="0" scrolling="no" />
              </Clearfix>
              <Clearfix visibleSmBlock>
                <iframe
                  title="small calendar"
                  src="https://www.google.com/calendar/embed?showTitle=0&amp;showTabs=0&amp;showCalendars=0&amp;showPrint=0&amp;height=600&amp;wkst=1&amp;bgcolor=%23FFFFFF&amp;src=fairviewrobotics%40gmail.com&amp;color=%23711616&amp;ctz=America%2FDenver"
                  className="calendar" width="100%" height="400" frameBorder="0" scrolling="no" />
              </Clearfix>
              <Clearfix visibleMdBlock>
                <iframe
                  title="medium calendar"
                  src="https://www.google.com/calendar/embed?showTitle=0&amp;showTabs=0&amp;showCalendars=0&amp;showPrint=0&amp;height=600&amp;wkst=1&amp;bgcolor=%23FFFFFF&amp;src=fairviewrobotics%40gmail.com&amp;color=%23711616&amp;ctz=America%2FDenver"
                  className="calendar" width="90%" height="400" frameBorder="0" scrolling="no" />
              </Clearfix>
              <Clearfix visibleLgBlock>
                <iframe
                  title="large calendar"
                  src="https://www.google.com/calendar/embed?showTitle=0&amp;showTabs=0&amp;showCalendars=0&amp;showPrint=0&amp;height=600&amp;wkst=1&amp;bgcolor=%23FFFFFF&amp;src=fairviewrobotics%40gmail.com&amp;color=%23711616&amp;ctz=America%2FDenver"
                  className="calendar" width="80%" height="500" frameBorder="0" scrolling="no" />
              </Clearfix>
            </Grid>
          </div>
        </div>
      </div>
    );
  }
}
