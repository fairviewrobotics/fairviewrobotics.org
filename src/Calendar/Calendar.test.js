import React from "react";
import { shallow } from "enzyme";
import Calendar from "./Calendar";

describe('Calendar', () => {
  it('renders without crashing', () => {
    shallow(
      <Calendar/>
    );
  });

  it('renders a Calendar page', () => {
    const wrapper = shallow(
      <Calendar/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});