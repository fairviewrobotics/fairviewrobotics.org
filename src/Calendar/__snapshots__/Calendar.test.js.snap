// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar renders a Calendar page 1`] = `
<Page
  backgroundSrc="6.jpg"
  title="Calendar"
>
  <div
    className="calendarBody"
  >
    <Grid
      bsClass="container"
      className="calendarContainer"
      componentClass="div"
      fluid={false}
    >
      <h2
        className="header"
      >
        Upcoming Events
      </h2>
      <Clearfix
        bsClass="clearfix"
        componentClass="div"
        visibleXsBlock={true}
      >
        <iframe
          className="calendar"
          frameBorder="0"
          height="300"
          scrolling="no"
          src="https://www.google.com/calendar/embed?showTitle=0&showNav=0&showDate=0&showPrint=0&showTabs=0&showCalendars=0&showTz=0&mode=AGENDA&height=600&wkst=1&bgcolor=%23FFFFFF&src=fairviewrobotics%40gmail.com&color=%23060D5E&ctz=America%2FDenver"
          title="extra small calendar"
          width="100%"
        />
      </Clearfix>
      <Clearfix
        bsClass="clearfix"
        componentClass="div"
        visibleSmBlock={true}
      >
        <iframe
          className="calendar"
          frameBorder="0"
          height="400"
          scrolling="no"
          src="https://www.google.com/calendar/embed?showTitle=0&showTabs=0&showCalendars=0&showPrint=0&height=600&wkst=1&bgcolor=%23FFFFFF&src=fairviewrobotics%40gmail.com&color=%23711616&ctz=America%2FDenver"
          title="small calendar"
          width="100%"
        />
      </Clearfix>
      <Clearfix
        bsClass="clearfix"
        componentClass="div"
        visibleMdBlock={true}
      >
        <iframe
          className="calendar"
          frameBorder="0"
          height="400"
          scrolling="no"
          src="https://www.google.com/calendar/embed?showTitle=0&showTabs=0&showCalendars=0&showPrint=0&height=600&wkst=1&bgcolor=%23FFFFFF&src=fairviewrobotics%40gmail.com&color=%23711616&ctz=America%2FDenver"
          title="medium calendar"
          width="90%"
        />
      </Clearfix>
      <Clearfix
        bsClass="clearfix"
        componentClass="div"
        visibleLgBlock={true}
      >
        <iframe
          className="calendar"
          frameBorder="0"
          height="500"
          scrolling="no"
          src="https://www.google.com/calendar/embed?showTitle=0&showTabs=0&showCalendars=0&showPrint=0&height=600&wkst=1&bgcolor=%23FFFFFF&src=fairviewrobotics%40gmail.com&color=%23711616&ctz=America%2FDenver"
          title="large calendar"
          width="80%"
        />
      </Clearfix>
    </Grid>
  </div>
</Page>
`;
